### PR TITLE
fix apk file open fails in some cases

### PR DESCRIPTION
**Pull request by tearf001**
**1. merge `getOpenFileIntent` and `validateIntent` to `validatedFileIntent`**
  when to comsume it,check the returning Indent is not null then run `startAcitvity(validatedFileIntent)`      
2.In `validatedFileIntent` method,     
if original contentType fails `validateIntent`, use `URLConnection.guessContentTypeFromStream`(inputFile) && `URLConnection.guessContentTypeFromName(path)` to get workable `mimeType`

### CODE REVIEW

File: android/src/main/java/vn/hunghd/flutterdownloader/IntentUtils.java [hunk 1]
```diff
@@ -39,7 +39,7 @@ public static synchronized Intent validatedFileIntent(Context context, String pa
         try {
             FileInputStream inputFile = new FileInputStream(path);
```
Comment (hnvn) [inline android/src/main/java/vn/hunghd/flutterdownloader/IntentUtils.java]: `FileInputStream` is needed to `close` at the end

File: android/src/main/java/vn/hunghd/flutterdownloader/IntentUtils.java [hunk 2]
```diff
@@ -38,9 +38,10 @@ public static synchronized Intent validatedFileIntent(Context context, String pa
         String mime = null;
         try {
             FileInputStream inputFile = new FileInputStream(path);
-            mime = URLConnection.guessContentTypeFromStream(inputFile);
+            mime = URLConnection.guessContentTypeFromStream(inputFile);//does not work in some target sdk version
             if(mime==null){
-                mime = URLConnection.guessContentTypeFromStream(new BufferedInputStream(inputFile));
+                mime = URLConnection.guessContentTypeFromName(path);//works fine
```
Comment (hnvn) [inline android/src/main/java/vn/hunghd/flutterdownloader/IntentUtils.java]: if `URLConnection.guessContentTypeFromStream` thows an exception, the below codes will be never executed

**Comment by tearf001** [comment]
I have read a lots of documents about android/java recently,
it's hard to read and write so many java codes well,sorry about it.

**Comment by hnvn** [comment]
It's OK, looks better, I will refactor some codes for you

**Comment by hnvn** [comment]
I've just updated your codes, let check it out and make sure it passed your case

**Pull request merged**